test/e2e/cluster/cluster.go

```diff
@@ -0,0 +1,323 @@
+/*
+Copyright 2018 Pressinfra SRL
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cluster
+
+import (
+	"fmt"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	. "github.com/onsi/gomega/gstruct"
+	gtypes "github.com/onsi/gomega/types"
+	core "k8s.io/api/core/v1"
+	meta "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	api "github.com/presslabs/mysql-operator/pkg/apis/mysql/v1alpha1"
+	orc "github.com/presslabs/mysql-operator/pkg/util/orchestrator"
+	"github.com/presslabs/mysql-operator/test/e2e/framework"
+	tutil "github.com/presslabs/mysql-operator/test/e2e/util"
+)
+
+const (
+	POLLING = 2 * time.Second
+)
+
+var _ = Describe("Mysql cluster tests", func() {
+	f := framework.NewFramework("mc-1")
+
+	It("scale up a cluster", func() {
+		// create a cluster
```

Comment:
All test start with a cluster. You should move that to a `BeforeEach`. Your `BeforeEach` should look like:
```go
name = fmt.Sprintf("cluster-%d", rand.Int31())

By("creating a new cluster secret")
pw = fmt.Sprintf("pw-%d", rand.Int31())
secret = newClusterSecret(name, f.Namespace.Name, pw)
_, err := f.ClientSet.CoreV1().Secrets(f.Namespace.Name).Create(secret)
Expect(err).ToNot(HaveOccurred(), "Failed to create secret '%s'", secret.Name)

By("creating a new cluster")
cluster = newCluster(name, f.Namespace.Name)
cluster, err = f.MyClientSet.MysqlV1alpha1().MysqlClusters(f.Namespace.Name).Create(cluster)
Expect(err).ToNot(HaveOccurred(), "Failed to create cluster: '%s'", cluster.Name)

By("testing the cluster readiness")
testClusterIsReady(f, cluster)

By("testing that cluster is registered with orchestrator")
testClusterIsRegistredWithOrchestrator()

cluster, err = f.MyClientSet.MysqlV1alpha1().MysqlClusters(f.Namespace.Name).Get(cluster.Name, meta.GetOptions{})
Expect(err).ToNot(HaveOccurred(), "Failed to get cluster %s", cluster.Name)
```